package/kedro_viz/models/flowchart.py

```diff
@@ -488,7 +488,12 @@ def is_tracking_node(self):
 
     def is_preview_node(self):
         """Checks if the current node has a preview"""
-        return hasattr(self.kedro_obj, "_preview")
+        metadata = getattr(self.kedro_obj, "metadata", {}) or {}
+        return bool(metadata.get("kedro-viz", {}).get("preview"))
```

Comment:
It feels like it shouldn't be that complicated. 
Many questions here:
* Does `preview: 0` count as `True` or `False`? It will be `False` now, then the `get("preview", 0) path will never be executed.

The same pattern accessing the nested dict appears in multiple places, maybe refactor it with a small util function